CONTRIBUTING.md

```diff
@@ -1,3 +1,119 @@
 # License
 
 By contributing your code, you agree to license your contribution under the MIT License.
+
+You also need to sign the [CLA](https://cla-assistant.io/bountysource/core) before your Pull Request is accepted.
+When you send your PR check the Pull Request page, it wil have a Link to signing the CLA.
+
+# How to Contribute
+Keep in mind this standardized setup is a work in progress, if you have any trouble please open an issue on the issue tracker and give as much info as possible.
+
+First of all you need [Docker](https://www.docker.com/what-docker) and [Docker Compose](https://docs.docker.com/compose/install/) installed.
+
+## Environment
+Bountysource uses Environment Variables for a lot of things. There's a suggested `.env.dev` file at the root directory that is loaded by the `docker-compose.yml`.
+
+It's recommended that you take a good look at the entire `.env.dev` file and, if necessary, change it to suit your needs, but at the very least you should pay attention to the following variables:
+
+0. URL for Buntysource API: `BOUNTYSOURCE_API_URL`
+0. URL for Bountysource Website (Front-End): `BOUNTYSOURCE_WWW_URL`
+0. URL for the Database: `DATABASE_URL`
+0. Environment for Rails and Rake: `RAILS_ENV` and `RAKE_ENV`, respectively.
+
+### About the `DATABASE_URL`
+This is a url, in the usual form `protocol://user:password@host/path`, just without the `password` since the postgres image doesn't use a password.
+The `path` part is the database name and you can change that without any issues, but before changing anything else keep in mind that the `host` matches the service in the `docker-compose.yml` and `docker-compose-test.yml` files, and the `user` matches the default on the `postgres` docker image.
+``` bash
+DATABASE_URL=postgres://postgres@pgsql/bountysource_dev
+```
+
+### The URLs
+Remember to keep the `API` and `WWW` URLs accessible from Outside Docker, either by changing your `hosts` file or leaving them as `0.0.0.0`:
+``` bash
+BOUNTYSOURCE_API_URL=http://0.0.0.0:3000/
+BOUNTYSOURCE_WWW_URL=http://0.0.0.0:3000/
+```
+
+## Standardized Setup for First-time use
+
+There is some setup and configuration you need to do before the project will run, this section covers those.
+
+Mainly, you need to create the Sphinx configuration file as well as Symlink it to its expected location before running the Search Daemon (Default command Sphinx image runs). You also have to generate the database and search indexes.
+
+0. Build container and start database.  
```

Comment:
It's a trade-off. I like the Rails solution of having a separate repo and I considered nesting Docker Images which is cleaner that this, but both will take a while and we need to have some way for people to get up and running.

Considering that, the easiest way seems to be Docker, but obviously we can look at other options, the PR was so I could get feedback. With Docker though there's no need for worrying about Ruby Versions/RVM, no need to install PostGres or Sphinx, just install latest Docker/Docker-Compose and run 5 commands.

And I would actually like to take those 5 commands down to a single one, but decided to PR before that so we could discuss it.

Do you think I should change it to RVM/Sphinx/Database (PostGres or maybe SQLite since it's easier, but I have to check if it will work...).